pkg/assembler/backends/arangodb/backend.go

```diff
@@ -157,231 +176,11 @@ func getBackend(ctx context.Context, args backends.BackendArgs) (backends.Backen
-			return nil, fmt.Errorf("failed to generate index for pointOfContact: %w", err)
-		}
-
-		// index for pointOfContact - Package
-		if err := createIndexPerCollection(ctx, db, pointOfContactStr, []string{"packageID", "email", "info", "since", "justification", "origin"}, false, "pointOfContactPackageID"); err != nil {
-			return nil, fmt.Errorf("failed to generate index for pointOfContact: %w", err)
-		}
-
-		// index for pointOfContact - Source
-		if err := createIndexPerCollection(ctx, db, pointOfContactStr, []string{"sourceID", "email", "info", "since", "justification", "origin"}, false, "pointOfContactSourceID"); err != nil {
-			return nil, fmt.Errorf("failed to generate index for pointOfContact: %w", err)
-		}
-
-		// index for vulnEqual
-		if err := createIndexPerCollection(ctx, db, vulnEqualsStr, []string{"vulnerabilityID", "equalVulnerabilityID", "justification", "origin"}, true, "vulnEqual"); err != nil {
-			return nil, fmt.Errorf("failed to generate index for vulnEqual: %w", err)
-		}
-
-		// index for vulnMetadata
-		if err := createIndexPerCollection(ctx, db, vulnMetadataStr, []string{"vulnerabilityID", "scoreType", "scoreValue", "timestamp", "origin"}, true, "vulnMetadata"); err != nil {
-			return nil, fmt.Errorf("failed to generate index for vulnMetadata: %w", err)
-		}
-
-		// GUAC key indices for package
-		if err := createIndexPerCollection(ctx, db, pkgNamespacesStr, []string{"guacKey"}, true, "byNsGuacKey"); err != nil {
-			return nil, fmt.Errorf("failed to generate guackey index for pkgNamespaces: %w", err)
-		}
-
-		if err := createIndexPerCollection(ctx, db, pkgNamesStr, []string{"guacKey"}, true, "byNameGuacKey"); err != nil {
-			return nil, fmt.Errorf("failed to generate guackey index for pkgNames: %w", err)
-		}
-
-		if err := createIndexPerCollection(ctx, db, pkgVersionsStr, []string{"guacKey"}, true, "byVersionGuacKey"); err != nil {
-			return nil, fmt.Errorf("failed to generate guackey index for pkgVersions: %w", err)
-		}
-
-		// GUAC key indices for source
-		if err := createIndexPerCollection(ctx, db, srcNamespacesStr, []string{"guacKey"}, true, "byNsGuacKey"); err != nil {
-			return nil, fmt.Errorf("failed to generate guackey index for srcNamespaces: %w", err)
-		}
-
-		if err := createIndexPerCollection(ctx, db, srcNamesStr, []string{"guacKey"}, true, "byNameGuacKey"); err != nil {
-			return nil, fmt.Errorf("failed to generate guackey index for srcNames: %w", err)
-		}
-
-		// GUAC key indices for vulnerabilities
-		if err := createIndexPerCollection(ctx, db, vulnerabilitiesStr, []string{"guacKey"}, false, "byVulnGuacKey"); err != nil {
-			return nil, fmt.Errorf("failed to generate index for vulnerabilities: %w", err)
+		fullCollectionIndexMap := getCollectionIndexMap()
+		for collectionName, indexes := range fullCollectionIndexMap {
+			for _, index := range indexes {
+				createIndexPerCollection(ctx, db, collectionName, index.fields, index.unique, index.name)
```

Comment:
Should we check error here?

Reply:
yes, @semmet95 please fix all the lint issues.

Reply:
Hi @mihaimaruseac 
you are correct, we should be checking the returned errors.
I have updated all such function calls to handle the returned error.